### PR TITLE
User story 23

### LINKED ISSUE #13: User Story 23, (EXTENSION 3/4) Book Index Page statistics shows top authors
```
As a Visitor,
When I visit the book index page,
In the Statistics area of the page,
I should also see three authors who have the highest-rated books
(average rating of all books for that author)
```

DO THESE STORIES FIRST:
- User Story 6, Book Index Page

### Checklist:

- [x] base code tests are written
- [x] base code is written to pass tests
- [x] edge case ("sad path") tests written
- [x] edge case code written
- [x] code is reviewed
- [x] branch is merged
- [x] all tests still pass

child of #5

child of #33

depends on #25

### PR DESCRIPTION
**Pull request by DavisC0801**
This PR...
Adds top author functionality to the book index page statistics section.
Adds additional author model methods to fetch average rating of all books.
resolves #13 (Book index page shows top authors)

### CODE REVIEW

File: app/models/author.rb [hunk 1]
```diff
@@ -7,4 +7,19 @@ class Author < ApplicationRecord
...
+    Author.joins(books: :reviews)
+    .order("AVG(reviews.rating) DESC")
+    .order(:name)
+    .select("authors.*")
+    .group("authors.id")
+    .limit(number)
+  end
+
+  def rating_percentage
+    (books.joins(:reviews)
```
Comment (chakeresa) [inline app/models/author.rb]:
```suggestion
    (books.joins(:reviews).average("reviews.rating") / 5.0 * 100).round(0).to_i
```

File: app/models/author.rb [hunk 2]
```diff
@@ -7,4 +7,19 @@ class Author < ApplicationRecord
...
+    .average("reviews.rating") / 5.0 * 100)
```
Comment (chakeresa) [inline app/models/author.rb]:
```suggestion
```

File: app/models/author.rb [hunk 3]
```diff
@@ -7,4 +7,19 @@ class Author < ApplicationRecord
...
+    .round(0).to_i
```
Comment (chakeresa) [inline app/models/author.rb]:
```suggestion
```

**Comment by chakeresa** [comment]
Looks good, just one comment.
